pkg/kubelet/kubelet_pods.go

```diff
@@ -607,7 +607,7 @@ func (kl *Kubelet) makeEnvironmentVariables(pod *v1.Pod, container *v1.Container
 
 // podFieldSelectorRuntimeValue returns the runtime value of the given
 // selector for a pod.
-func (kl *Kubelet) podFieldSelectorRuntimeValue(fs *v1.ObjectFieldSelector, pod *v1.Pod, podIP string) (string, error) {
+func (kl *Kubelet) podFieldSelectorRuntimeValue(fs *v1.ObjectFieldSelector, pod *v1.Pod, podIP, hostIP string) (string, error) {
```

Comment:
Yeah I had the same thought. I was told there are some race conditions with podIP so I decided to define it in the same place as podIP (also defining in the same place seemed to make more sense in terms of discoverability). 

Thinking about it now I realize there are no possibility of race conditions with hostIP like there would be with podIP so calling `GetHostIP` further down does make sense. There's a lot of refactoring to do here anyways, I figured I implement it in the most straight forward way I can and refactor this later on. What do you think?